src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node head;
+    private Node tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (newNode.prev != null) {
+            newNode.prev.next = newNode;
+        }
+        if (head == null) {
+            head = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index out of range");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node currentNode = findNode(index);
+        if (index == 0) {
+            Node newNode = new Node(null, value, currentNode);
+            currentNode.prev = newNode;
+            head = newNode;
+        } else {
+            Node newNode = new Node(currentNode.prev, value, currentNode);
+            currentNode.prev.next = newNode;
```

Comment:
```suggestion
        if (index == 0) {
            Node newNode = new Node(null, value, head);
            head.prev = newNode;
            head = newNode;
        } else {
            Node currentNode = findNode(index);
            Node newNode = new Node(currentNode.prev, value, currentNode);
            currentNode.prev.next = newNode;
```
no need to look for a node if you know that its head